### PR TITLE
Fix and Refactor NonAVX512 CPU platform graceful exit for BFloat16 Ops with eigen fallback

### PR DESCRIPTION
**Pull request by nammbash**
Attn: @penpornk 
The Bug in PR https://github.com/tensorflow/tensorflow/pull/40212 which addresses NONAVX512 CPU Platform graceful exit, by removing the check for (T == DT_BFLOAT16) is fixed and refactored for clear understanding.

### **Problem:** 
( in https://github.com/tensorflow/tensorflow/pull/40212 ) Removed from review comments
`return (T == DT_FLOAT || (T == DT_BFLOAT16 && CheckBfloat16Support(T)));`
to
`return (T == DT_FLOAT || CheckBfloat16Support(T));`
created lot of failures internally.

### **Fix**
1. Put back the datatype check. `T == DT_BFLOAT16`
2. Refactored the code for clear understanding by making the function not Datatype dependent. `IsBF16SupportedByOneDNNOnThisCPU()`
3. For unsupported mkl bfloat16 ops, we fall back to Eigen otherwise
4. Created separate function for the warning `BF16UnsupportedWarning()`

### CODE REVIEW

File: tensorflow/core/graph/mkl_graph_util.h [hunk 1]
```diff
@@ -216,7 +224,18 @@ static inline bool IsMklNameChangeOp(const string& op_name, DataType T) {
...
-    isTypeAllowed = (isTypeAllowed || CheckBfloat16Support(T));
+    if (!isTypeAllowed) {
+      if (T == DT_BFLOAT16) {
+        if (IsBF16SupportedByOneDNNOnThisCPU()) {
+          isTypeAllowed = true;
+        } else {
+          // Restrict bfloat16 ops to platforms with at least AVX512 support,
+          // fall back to Eigen implementation otherwise.
+          BF16UnsupportedWarning();
+          isTypeAllowed = false;
```
Comment (penpornk) [inline tensorflow/core/graph/mkl_graph_util.h:235]: `isTypeAllowed` is already `false`. So this line is not necessary. Would you like to remove it? (I'm also fine leaving it in to save another round of testing, given that we are close to the branch cut deadline.)

**Review by penpornk — COMMENTED** [review]
Thank you for the PR!

File: tensorflow/core/graph/mkl_graph_util.h (hunk 1 above)
Comment (nammbash) [reply tensorflow/core/graph/mkl_graph_util.h:235]: Yes, was double qualifying just to be sure this time around. :). Let us move on this time in the interest of getting the CI working internally and for the deadline. Will make a note internally for refactoring in future.

Comment (nammbash) [reply tensorflow/core/graph/mkl_graph_util.h:235]: Thank you for looking into this very soon! @penpornk

**Review by penpornk — APPROVED** [review]
Thank you for the quick reply! Let's get this in then. :)

**Pull request merged**